### ISSUE TITLE
NEC - PC-FX: Two BIOS listed with the same hash info.

### ISSUE THREAD

**Issue by Whovian9369** [libretro/libretro-database#373]
`rom ( name pcfx.bios size 1048576 crc 76ffb97a md5 08e36edbea28a017f79f8d4f7ff9b6d7 sha1 1a77fd83e337f906aecab27a1604db064cf10074 )`
`rom ( name pcfxbios.bin size 1048576 crc 76ffb97a md5 08e36edbea28a017f79f8d4f7ff9b6d7 sha1 1a77fd83e337f906aecab27a1604db064cf10074 )`

**Comment by markwkidd** [comment]
Hmm this seems like an error with the libretro DAT.

Mednafen PC-FX only requires one BIOS file, which I note has the same checksum as the duplicated BIOS entry in the libretro database: http://mednafen.fobby.net/documentation/pcfx.html#Section_firmware_bios

My speculation is that either pcfx.bios is required, or pcfxbios.bin. There was an issue related to this (https://github.com/libretro/libretro-database/issues/345) and a subsequent PR (https://github.com/libretro/libretro-database/pull/346).

@lasers and @retro-wertz were involved in this prior effort to fix the BIOS information. Maybe one of them will be able to resolve this.

**Comment by retro-wertz** [comment]
pcfxbios.bin and pcfxv101.bin are the filename format default to mame, they can probably be removed since the system is not working there yet anyways.

pcfx.rom/pcfx.bios is used by mednafen - correct working CRC is 76ffb97a

**Comment by lasers** [comment]
Firmware checking was recently added. I put it there when I grepped the directories looking for potential missing bios (bios|bin) to be added in the database.

MAME + Mednafen = Different cores requiring same file (?) under different names (?).

If MAME (pcfx) isn't working right now, go ahead and remove `pcfxbios.bin`, `pcfxv101.bin` (or comment out). See if we want to keep or remove `fx-scsi.rom` as retro-wertz say it's a feature for developers. `pcfxga.rom` too?

Initial Issue https://github.com/libretro/libretro-database/issues/351

MAME
└── https://github.com/libretro/mame/blob/master/src/mame/drivers/pcfx.cpp#L503-L512

Mednafen PCFX
├── https://github.com/libretro/beetle-pcfx-libretro/blob/master/libretro.cpp#L432-L440
└── https://github.com/libretro/beetle-pcfx-libretro/blob/master/libretro.cpp#L989-L990

**Comment by retro-wertz** [comment]
Mame's pcfxbios.bin and Mednafen's pcfx.bios is the same rom(crc matched), just different filenames
i think its a mistake when the initial version of PCFX core was ported, "pcfx.bios" is suppose to be a variable-name while rom name is suppose to be "pcfx.rom" 

with *Check for Missing Firmware Before Loading=ON":
-current mame, mame2014, mame2003 works without pcfxbios.bin and pcfxv101.bin
-mednafen core works without fx-scsi.rom
(i normally do have these 3 bios filenames anywhere in retroarch)

**Comment by markwkidd** [comment]
**About the MAME PC-FX BIOSes**

pcfxbios.bin, pcfxv101.bin, pgfxa.rom, and fx-scsi.rom are all mentioned by name and CRC in the MAME pcfx driver: https://github.com/mamedev/mame/blob/master/src/mame/drivers/pcfx.cpp#L503-L512

- MAME is aware of two BIOS revisions, **pcvfbios.bin = BIOS 1.00 and pcfxv101.bin = BIOS 1.01**
- As previously mentioned, MAME is aware of **fx-scsi.rom** which is an original SCSI add-on from NEC (a page about that: http://nfgworld.com/mb/thread/416-NEC-s-FX-SCSI-Adaptor-for-PC-FX)
- MAME is also aware of **pgfxa.rom**, an original ISA expansion card from NEC http://www.progettoemma.net/mess/system.php?machine=pcfxga

**Comment by markwkidd** [comment]
About the Mednafen PC-FX BIOS

@lasers as you say, it seems that there was an error in porting the Mednafen PC-FX core which mistook the name of the configuration variable (pcfx.bios) for the name of the BIOS file (pcfx.rom)

The relevant section of the upstream Mednafen core is here: https://github.com/libretro/beetle-pcfx-libretro/blob/master/libretro.cpp#L989

I've even found a few forum posts from people moving back and forth from the standalone to the libretro core who have trouble with this filename issue.

**Comment by markwkidd** [comment]
I found the port error. It occurs on this line: https://github.com/libretro/beetle-pcfx-libretro/blob/master/mednafen/settings.cpp#L108

In the process of porting from standalone to libretro, the MDFN_GetSettingS function was simplified from the original source (which provides settings for all of the systems that Mednafen emulates). Probably due to an errant copy paste or other oversight, the variable name was substituted for the filename.

That isn't really relevant to this database/hashing Issue, so I suppose the thing to do is file a PR and see if the devs would like to fix the name before we tweak the DAT: https://github.com/libretro/beetle-pcfx-libretro/pull/15

**Comment by markwkidd** [comment]
OK, four PRs later I think this issue can be settled for now. :)

Moving forward, the Mednafen/Beetle PC-FX core will be looking for a BIOS file called **pcfx.rom** with size of 1,048,576 and a CRC of 76ffb97a (exact same as the BIOS with SHA1 1a77fd83e337f906aecab27a1604db064cf10074 recommended by Mednafen standalone).

The BIOS DATs have been updated, but there is still going to be a duplicate BIOS hash so we can have the filename that Mednafen uses and the one that the mamedev source code is looking for. There are also some other BIOS files (details on those in prior comments) which may be used by either or both of the emulator cores in different scenarios.

In summary, @Whovian9369 your issue surfaced an error in the port from the standalone Mednafen emulator to libretro. So even though there will still be a duplicate hash in the BIOS dat, it will be for the correct filename! Thanks.
